src/balances/balances.service.ts

```diff
@@ -83,7 +83,7 @@ export class BalancesService {
         name: txBalance.token?.name,
         logoUri: logoUri,
       },
-      balance: txBalance.balance,
+      balance: txBalance.balance.toString(),
```

Comment:
This is because of the type change. `balance` is now a `bigint`

Reply:
I was wondering if `txBalance.balance` could be undefined, in which case this would produce a runtime error. I assume it's ok to fail fast and loud in that scenario, but this leads me to another question, should we validate payloads coming from Config/Transaction services? That could be done in a separate PR, of course, but could provide a more controlled behaviour when an unexpected payload shape comes into the service. I've happily used Joi (https://joi.dev/) in the past for that, but Nest recommends this: https://github.com/typestack/class-validator

What do you think? 🙂

Reply:
> I was wondering if `txBalance.balance` could be undefined, in which case this would produce a runtime error. I assume it's ok to fail fast and loud in that scenario, but this leads me to another question, should we validate payloads coming from Config/Transaction services? That could be done in a separate PR, of course, but could provide a more controlled behaviour when an unexpected payload shape comes into the service. I've happily used Joi (https://joi.dev/) in the past for that, but Nest recommends this: https://github.com/typestack/class-validator
> 
> What do you think? 🙂

I agree. We should look into these solutions for validating the incoming Json. Throwing another one into the list: https://ajv.js.org/